File: read-pdfs.py
import os
from pypdf import PdfReader

dir_path = "/Users/mabyprochilo/Desktop/Newsletter"
files = [f for f in os.listdir(dir_path) if f.endswith('.pdf')]
combined_text = ""

for i, f in enumerate(files[:3]):
    try:
        reader = PdfReader(os.path.join(dir_path, f))
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        combined_text += f"\n--- NEWSLETTER {i+1} ---\n{text}\n-----------------------\n"
    except Exception as e:
        print(f"Error reading {f}: {e}")

with open("extracted_newsletters.txt", "w") as out:
    out.write(combined_text)
print("Done extracting text.")

File: scripts/read-pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function readPdfs() {
    const dir = '/Users/mabyprochilo/Desktop/Newsletter';
    const files = fs.readdirSync(dir).filter(f => f.endsWith('.pdf'));
    let combinedText = '';
    
    // Leggi i primi 3 PDF per non sovraccaricare
    for (let i = 0; i < Math.min(3, files.length); i++) {
        const filePath = `${dir}/${files[i]}`;
        const dataBuffer = fs.readFileSync(filePath);
        try {
            const data = await pdf(dataBuffer);
            combinedText += `\n--- NEWSLETTER ${i+1} ---\n` + data.text + `\n-----------------------\n`;
        } catch(e) {
            console.error('Error reading', files[i]);
        }
    }
    fs.writeFileSync('scripts/extracted_newsletters.txt', combinedText);
    console.log('Done extracting text. Saved to scripts/extracted_newsletters.txt');
}
readPdfs();
